sdk/component/internal/storer/local/sqlite/sqlite.go

```diff
@@ -50,31 +77,174 @@ func NewManager(dbPath string, dbName string) (*manager, error) {
 		return nil, fmt.Errorf("could not open sqlite db: %w", err)
 	}
 
-	return &manager{
+	mgr := &manager{
+		clock:  clockwork.NewRealClock(),
 		db:     db,
 		dbName: dbName,
-	}, nil
+	}
+
+	for _, opt := range opts {
+		if err := opt(mgr); err != nil {
+			return nil, fmt.Errorf("could not apply option: %w", err)
+		}
+	}
+
+	return mgr, nil
 }
 
-// TODO - implement me.
-func (m *manager) Read(ctx context.Context) ([]*ocsf.VulnerabilityFinding, error) {
-	return nil, nil
+// Validate. TODO - implement.
+func (m *manager) Validate(*ocsf.VulnerabilityFinding) error {
+	return nil
 }
 
-// TODO - implement me.
-func (m *manager) Write(ctx context.Context, findings []*ocsf.VulnerabilityFinding) error {
+// Read finds Vulnerability Findings by workflowID.
+// It returns storer.ErrNoFindingsFound is not vulnerabilities were found.
+func (m *manager) Read(ctx context.Context, workflowID uuid.UUID) ([]*ocsf.VulnerabilityFinding, error) {
+	stmt, err := m.db.PrepareContext(ctx, `
+		SELECT (findings) 
+			FROM finding 
+			WHERE workflow_id = :workflow_id
```

Comment:
looking at the code again, I think that what we actually mean `instance_id` instead of `workflow_id`. instance id is the UUID of each execution of a workflow. the workflow itself doesn't have an ID, it could have a name or some other identifier, but in the context of the OSS smithy it will be a file with the stages in the local filesystem